conflux/src/adapter.ts

```diff
@@ -0,0 +1,35 @@
+import { Requester, Validator, AdapterError } from '@chainlink/external-adapter'
+import { Config, ExecuteWithConfig, ExecuteFactory } from '@chainlink/types'
+import { makeConfig, DEFAULT_ENDPOINT } from './config'
+import { conflux } from './endpoint'
+
+const inputParams = {
+  endpoint: false,
+}
+
+export const execute: ExecuteWithConfig<Config> = async (request, config) => {
+  const validator = new Validator(request, inputParams)
+  if (validator.error) throw validator.error
+
+  Requester.logConfig(config)
+
+  const jobRunID = validator.validated.id
+  const endpoint = validator.validated.data.endpoint || DEFAULT_ENDPOINT
+
+  switch (endpoint) {
+    case conflux.NAME: {
+      return await conflux.execute(request, config)
+    }
+    default: {
+      throw new AdapterError({
+        jobRunID,
+        message: `Endpoint ${endpoint} not supported.`,
+        statusCode: 400,
+      })
+    }
+  }
+}
+
+export const makeExecute: ExecuteFactory<Config> = (config) => {
```

Comment:
The compiler would probably complain of having the custom `Config`. A current tweak is typing the resulting function only as `Execute`